connector-common/src/main/java/com/redhat/cloud/notifications/connector/EngineToConnectorRouteBuilder.java

```diff
@@ -72,6 +80,12 @@ public void configure() throws Exception {
                 .to(log(getClass().getName()).level("DEBUG").showHeaders(true).showBody(true))
                 .filter(incomingCloudEventFilter)
                 .process(this.incomingKafkaReinjectionHeadersProcessor)
+                .choice()
+                    .when(header(Constants.X_RH_NOTIFICATIONS_CONNECTOR_PAYLOAD_ID_HEADER).isNotNull())
+                        .process(this.payloadDetailsRequestPreparer)
+                        .to(http(this.connectorConfig.getNotificationsEngineHostname()))
+                        .process(this.payloadDetailsResponseProcessor)
```

Comment:
We could use a `rest-client` call from a Camel `Processor` instead of the [request preparer / perform call / response processor] Camel forces us to write.

Reply:
When we created this connector, we refused to use `rest-client` because we had no idea if it can have a impact on regurlar Camel http management. Do we have more documentation about it?
Shouldn't we keep it a more global refactoring later?

Reply:
The thing is that using the Rest Client does significantly reduce the code we need to write and the tests we need to make. I would vote for using it...